File: 290-Word-Pattern.cs
public class Solution {
    public bool WordPattern(string pattern, string s) {
        string[] words = s.Split(' ');
        if (pattern.Length != words.Length) {
            return false;
        }
        Dictionary<char, string> mapP = new Dictionary<char, string>();
        Dictionary<string, char> mapS = new Dictionary<string, char>();

        for (int i = 0; i < pattern.Length; i++) {
            if (mapP.ContainsKey(pattern[i])) {
                if (mapP[pattern[i]] != words[i]) {
                    return false;
                }
            } else {
                mapP[pattern[i]] = words[i];
            }

            if (mapS.ContainsKey(words[i])) {
                if (mapS[words[i]] != pattern[i]) {
                    return false;
                }
            } else {
                mapS[words[i]] = pattern[i];
            }
        }
        return true;
    }
}

File: 290-Word-Pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
    let words = s.split(' ');
    if (pattern.length !== words.length) {
        return false;
    }

    let mapP = new Map();
    let mapS = new Map();

    for (let i = 0; i < pattern.length; i++) {
        
        if (mapP.has(pattern[i])) {
            if (mapP.get(pattern[i]) !== words[i]) {
                return false;
            }
        } else {
            mapP.set(pattern[i], words[i]);
        }

        if (mapS.has(words[i])) {
            if (mapS.get(words[i]) !== pattern[i]) {
                return false;
            }
        } else {
            mapS.set(words[i], pattern[i]);
        }
    }

    return true;
};
